### DISCUSSION TITLE
What is the meaning of sizes in reposize?

### DISCUSSION THREAD

**Discussion by DianaNites** [rustic-rs/rustic#706] [Q&A] [answered]
They don't correspond to anything I can tell, its sure not the size of my B2 bucket, or the source data, so what are they?

Whats size vs "size in pack"? is there anywhere i can find the actual size of all data in the repo, in this case the size of the bucket its on?

**Comment by DianaNites** [comment, marked as answer]
Some investigation of my own reveals rustic reports size in GiB, but the Backblaze UI reports size in GB and does not reveal useful units, and additionally, absurdly, the only way to *get* the size list all the files and tally it yourself, only convienelty exposed through [`b2 get-bucket --showSize <bucket>`](https://github.com/Backblaze/B2_Command_Line_Tool/issues/404#issuecomment-373744409).

Which, notably, entails additional, charged, API calls.

When accounting for that, rounding on both sides, and file versions, the sizes now match

**Comment by aawsome** [comment]
Actually there are three things about the blobs saved in a repository:
- size of all blobs: This is reported in the "index" section under "Total" -> "Total Size". It is the raw size of all blobs saved in the repository
- The size these blobs need in the repository: This is reported in the "index" section under "Total" -> "Total Size in Packs". Note that this differs to the first information as the blobs are compressed (typically reducing the size) and encrypted (which adds a small overhead per blob).
- The acutal size of all packs. This is reported in the "repository files" section under "Pack" -> "Total Size". This size should equal the used size on the storage backend. This again is a bit higher than the second information as additionally to the compressed and encrypted blobs, also pack headers are saved in the pack files which also add a little overhead.
